aiohttp/client.py

```diff
@@ -322,6 +331,10 @@ def _request(self, method, url, *,
                         elif not scheme:
                             r_url = url.join(r_url)
 
+                        if url.origin() != r_url.origin():
```

Comment:
Side note: it seems like relative `Location` [is legit](https://tools.ietf.org/html/rfc7231#section-7.1.2) now.

Reply:
Relative URL is joined to original two lines above.
`r_url` is absolute at this line.

Reply:
I'm interpreted it as schemaless url rather then relative one. Fine then.